src/commands/project/pipeline.ts

```diff
@@ -76,6 +77,41 @@ export const checkDependencies = (projectPath: string): void => {
   }
 };
 
+/**
+ * Returns a git repository url
+ *
+ * @param opts Options object from commander.
+ * @param gitOriginUrl Git origin URL which is used to set values
+ *        for pipeline, repoName and repoUrl
+ */
+export const validateRepoUrl= (
+  opts: CommandOptions,
+  gitOriginUrl: string
+): string => {
+  try {
+    if (opts.repoUrl) {
+      return opts.repoUrl;
+    }
+    const repoUrl = getRepositoryUrl(gitOriginUrl);
+    if (!repoUrl) {
+        throw Error(`Repo url not defined. Are you in a spk project folder?`);
```

Comment:
this error is lost when we throw an exception here and then catch it later (at line 111) where it throws a new error.